libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu

```diff
@@ -927,6 +928,233 @@ namespace helpers {
+                auto scalesDim = yEnd - yStart;
+                auto yScaleCache = sharedPtr + scalesDim * y * sizeof(ScaleCache<T>);
+
+                //auto startPtr = sharedPtr + y * scalesDim * sizeof(float);
+                //float* yScales = yScalesShare + y * sizeof(float) * scalesDim;//reinterpret_cast<float*>(startPtr); //shared + y * scalesDim * y + scalesDim * sizeof(T const *) [scalesDim];
+                //T const** yPtrs = yPtrsShare + y * sizeof(T const*) * scalesDim; //[scalesDim];
+                //yPtrs = reinterpret_cast<T const**>(sharedBuf);
+                float* output = outputPtr + (batch * pSt->outHeight  +  y)  * pSt->channels * pSt->outWidth;
+                //int k = 0;
+                for (Nd4jLong i = yStart, k = 0; i < yEnd; ++i, ++k) {
+                    float scaleY;
+                    if (i < inY) {
+                        scaleY = (i + 1 > inY1 ? pSt->heightScale : i + 1 - inY);
+                    } else {
+                        scaleY = (i + 1 > inY1 ? inY1 - i : 1.0);
+                    }
+                    yScaleCache[k].yScale = scaleY;
+                    yScaleCache[k].yPtr = inputPtr + (batch * pSt->inHeight * pSt->inWidth * pSt->channels + bound(i, pSt->inHeight) * pSt->inWidth * pSt->channels);
+                }
+
+                if (pSt->channels == 3) {
+                    for (Nd4jLong x = 0; x < pSt->outWidth; ++x) {
+                        const CachedInterpolation& xCache = caches[x];
+                        computePatchSumOf3Channels<T>(scale, *pSt, yScaleCache, scalesDim, xCache, output);
+                        output += pSt->channels;
+                    }
+                } else {
+                    for (Nd4jLong x = 0; x < pSt->outWidth; ++x) {
+                        const CachedInterpolation &xCache = caches[x];
+                        computePatchSum<T>(scale, *pSt, yScaleCache, scalesDim, xCache, output);
+                        output += pSt->channels;
+                    }
+                }
+            }
+        }
+    }
+
+    template <typename T>
+    static void resizeArea(cudaStream_t* stream, ImageResizerState const& st, CachedInterpolation* cache,
+            NDArray const* input, NDArray* output) {
+
+        T const* inputPtr = reinterpret_cast<T const*>(input->getSpecialBuffer());
+//        float* yScales;
+//        T const** yPtrs;
+        float scale = 1.f / (st.heightScale * st.widthScale);
+        auto outputPtr = reinterpret_cast<float*>(output->specialBuffer()); // output is always float. TO DO: provide another float types also with  template <typename X, typename Z> declaration
+        ImageResizerState* pSt;
+        auto err = cudaMalloc(&pSt, sizeof(ImageResizerState));
+        err = cudaMemcpy(pSt, &st, sizeof(ImageResizerState), cudaMemcpyHostToDevice);
+
+        resizeAreaKernel<T><<<128, 1, 2048, *stream>>>(pSt, cache, scale, inputPtr, input->getSpecialShapeInfo(), outputPtr,
```

Comment:
Are we sure we can't make it work with more than 1 thread? :)

Reply:
we can use more then 1, but not free. Now it is 4